### ISSUE TITLE
How do you save the images TrackPy creates for reports 

### ISSUE THREAD

**Issue by Rory235** [soft-matter/trackpy#729]
I have tried the following 

plt.figure("1")
tp.plot_traj(t1, superimpose=frame[1000])
plt.imsave("C:/Users/Rory/Downloads", arr="MxN")

**Comment by nkeim** [comment]
Have you tried *not* calling `plt.figure()` beforehand?

If you want to remove all ambiguity, you can get or create axes in the current figure and make sure that `plot_traj` uses them. Something like (not sure if it's 100% correct):

```python
fig = plt.figure()
ax = fig.get_axes()
tp.plot_traj(..., ax=ax)
fig.savefig(...)

**Comment by Rory235** [comment]
So trackpy still creates the plot without the `plt.figure()`

I gave tried this 

`fig1=tp.plot_traj(t1, superimpose=frame[1000])
fig1.savefig(r'C:\Users\Rory\Downloads\1.png')`

but get the error 

`AttributeError                            Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_24356/3548345737.py in <module>
      1 fig1=tp.plot_traj(t1, superimpose=frame[1000])
----> 2 fig1.savefig(r'C:\Users\Rory\Downloads\1.png')

AttributeError: 'AxesSubplot' object has no attribute 'savefig'`

**Comment by nkeim** [comment]
That's right. `plot_traj` returns an Axes object, not a figure. So if you are being explicit about it, you should first get the figure, then tell `plot_traj` to use axes within it, and then save the figure.

I'll admit I'm a bit surprised that `tp.plot_traj(...); plt.savefig(filename)` doesn't work. I had thought that it was that simple.

**Comment by Rory235** [comment]
Ahhhh I see. I gave this a try 

`fig1 = plt.figure()
ax = fig1.get_axes()
tp.plot_traj(t1, ax=ax)
fig1.savefig(r'C:\Users\Rory\Downloads\1.png')`

getting 

`AttributeError: 'list' object has no attribute 'set_xlabel'`

**Comment by nkeim** [comment]
You're asking the wrong person for matplotlib help, but you could do this instead.

```python
fig = plt.figure()
ax = plt.gca()
tp.plot_traj(..., ax=ax)
fig.savefig(...)
```

or

```python
ax = tp.plot_traj(...)
ax.figure.savefig(...)
```

**Comment by vivarose** [comment]
The error happens because fig1.get_axes() returns a _list_ of all axes on your figure.

@nkeim's solution to get the axes should work, or you can do
`fig1, ax = plt.subplots()`
instead of plt.figure()

**Comment by vivarose** [comment]
Also, I recommend saving the image as a pdf. Then it will be a vectorized image.
Sometimes `bbox_inches='tight'` is helpful to ensure that the bounding box goes around the image.
